accepted/0000-sle-micro-reboot-needed.md

```diff
@@ -0,0 +1,61 @@
+- Feature Name: SLE Micro: report when a reboot is needed
+- Start Date: 2021-09-23
+
+# Summary
+[summary]: #summary
+
+After a system level change in SLE Micro, SUSE Manager should indicate in the UI when a reboot is needed.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, for SLE Micro, we display the following message on the detail page: `OS with transactional updates: please reboot after any packages and/or channel changes`.
+
+This is not ideal when it comes to usability. The user is responsible for knowing when a reboot is needed, and scheduling those reboots.
+
+
+# Detailed design
+[design]: #detailed-design
+
+The SLE Micro uses transactional updates. Transactional updates are atomic (all updates are applied only if all updates succeed) and support rollbacks. They don't affect a running system as no changes are activated until after the system is rebooted.
+To be able to have up-to-date information regarding reboot needed, we need to update this information after every state apply.
+For that, this proposed solution is based on the following steps:
+ - Create an execution module to check if a reboot is required
+ - After every state apply run this execution module
+ - Store the result in a postgresql column
+ - Report reboot needed in the UI based on this column
```

Comment:
There is another module that can help (I am not sure if the service is present in SLE Micro) when doing a reboot.

[reboot_manager](https://docs.saltproject.io/en/3004/ref/modules/all/salt.modules.rebootmgr.html) is also supported in salt.  With this we can create "reboot windows" parameterized by the user.

We can trigger a reboot (using the configured option by the user) using the [reboot](https://docs.saltproject.io/en/3004/ref/modules/all/salt.modules.transactional_update.html#salt.modules.transactional_update.reboot) command from the `transactional_update` module.

Reply:
That's true, and it's a good way if we can assume that `rebootmgr` is the only reboot manager used. As far as I am aware, `transactional-update` can schedule reboots for 4 reboot managers (https://github.com/openSUSE/transactional-update/blob/2e15fee623b24e9ab8e382253d4930e91a912bc8/lib/Reboot.cpp#L30) which we probably all need to support in Uyuni. We could get around this by defining that SLE Micro managed by Uyuni can only use reboot manager X, which could also be Uyuni itself (i.e. transactional-update would need to be expanded to trigger a reboot for a 5th reboot manager)

Reply:
Right, and the method is set in [transactional-update.conf](https://github.com/openSUSE/transactional-update/blob/master/man/transactional-update.conf.5.xml#L76), so can be configured by Salt.

That is also with the [reboot](https://docs.saltproject.io/en/3004/ref/modules/all/salt.modules.transactional_update.html#salt.modules.transactional_update.reboot) command, together with [pending_transaction](https://docs.saltproject.io/en/3004/ref/modules/all/salt.modules.transactional_update.html#salt.modules.transactional_update.pending_transaction) command pair should be used.